src/pages/blog/performance-update.md

```diff
@@ -0,0 +1,57 @@
+
+### One size, does not fit everyone ###
+
+Ironic fulfills the needs of several different use cases. From a Systems Adminsitrator just needing to deploy the same OS image to a number of machines in fairly rapid order. To software deploying specialized ramdisk iamges and configuration such as those that may be used to stand up a small private cloud. To the largest and most complex deployments where Ironic powers the management of all Bare Metal nodes and resources within an organization.
+
+Not only does the scale differ, but so do the usage patterns, and ultimately the resources required to continue to manage a dynamic Bare Metal cloud. In this exercise, this was something we came to learn with our code.
+
+We treated database queries as one size fits all, and to put it bluntl, they cannot be treated as such. Without witing more on the subject than anyone would want to read, there are trade-offs. Ironic now attempts to leverage the trade-off when lists of nodes are being requested which *should* improve general node list performance. The trade off being in those cases, Ironic will no longer use a database join to pull in a composite field for a node. This is because that join was actually triggering additional overhead in the SQL client as the result set was being downloaded and converted for use by the program. Basic performance testing shows this change to be on-par with when we utilzied joins, but it should help operators who use one or more node traits assigned to a node.
+
+## Next Step ##
+
+Now that we have have some awesome performance improvement merged to Ironic, there are obviously several future steps we *can* take. Only time will tell if we *will* take them.
+<br>
+* Make **chassis_uuid** use a join - Presently, explicitly asking for a node's chassis_uuid is kind of slow. Unfortunately this is an object model change and cannot be backported.
+* Make **allocation_uuid** use a join - Explicitly asking for an allocation, also causes some additional overhead query. Unfortunately, this is also an object model change and cannot be backported.
+* Improve Nova's interaction with nova.virt.ironic - This is not explicitly in Ironic, but the driver for Ironic in nova is subjected to some calls internally that make sense when your nova-compute service is a hypervisor, and not a transport through to Ironic. Specifically [bug 1923955](https://bugs.launchpad.net/nova/+bug/1933955) is one of those cases. We believe this can use the internal cache.
+* We may explore our periodic task handling, but at present this does not feel like it would have much of an operational gain for users of Ironic.
+<br>
+If your hoping a list including the node conductor can see a performance improvement, I wouldn't expect that unless we consider additional data model changes. But if you have a use case, please bring it to our attention and maybe it will inspire us!
```

Comment:
node conductor? That doesn't make sense.

Reply:
Revising.